root/static/scripts/edit/components/forms.js

```diff
@@ -59,3 +59,36 @@ MB.initializeArtistCredit = function (form, initialArtistCredit) {
     );
   });
 };
+
+/*
+ * Registers a beforeunload event listener on the window that prompts
+ * the user if any of the page's form inputs have been changed.
+ */
+MB.installFormUnloadWarning = function () {
+  let modified = false;
+
+  const form = document.querySelector('#page form');
+
+  /*
+   * This is somewhat heavy-handed, in that it will still warn even if the
+   * user changes an input back to its original value.
+   */
+  form.addEventListener('change', () => {
+    modified = true;
+  });
```

Comment:
Does the relationship editor e.g. emit custom events for changes that I could also listen for here?

Reply:
No custom events, but...perhaps if you check `#relationship-editor` for the presence of hidden inputs, that would indicate whether changes were made.

Reply:
Hmm. That's a clever idea, but I'm not seeing any hidden relationship inputs at the time of the `beforeunload` event. This is my first time glancing at the relationship editor code, but is it possible that `root/static/scripts/relationship-editor/utility/prepareHtmlFormSubmission.js` only adds the hidden inputs immediately before the form is posted? If so, I don't think I'll be able to use that approach here.

Reply:
Ah, that's right, it only adds them once the submit event is triggered. Sorry.